### PR TITLE
cleanup advanced failsafe support

### PR DESCRIPTION
**Pull request by tridge**
this is in preparation for adding AFS support to copter.

### CODE REVIEW

File: libraries/APM_OBC/APM_OBC.cpp [hunk 1]
```diff
@@ -30,6 +30,12 @@ extern const AP_HAL::HAL& hal;
 
 // table of user settable parameters
 const AP_Param::GroupInfo APM_OBC::var_info[] = {
+    // @Param: ENABLE
+    // @DisplayName: Enable Advanced Failsafe
+    // @Description: This enables the advanced failsafe system. If this is set to zero (disable) then all the other AFS options have no effect
+    // @User: Advanced
+    AP_GROUPINFO_FLAGS("ENABLE",       11, APM_OBC, _enable, 0, AP_PARAM_FLAG_ENABLE),
```
Comment (lucasdemarchi) [inline libraries/APM_OBC/APM_OBC.cpp]: is 11 correct here?

File: libraries/AP_AdvancedFailsafe/AP_AdvancedFailsafe.cpp [hunk 2]
```diff
@@ -15,142 +15,142 @@
  */
 
 /*
-    APM_OBC.cpp
+    AP_AdvancedFailsafe.cpp
```
Comment (lucasdemarchi) [inline libraries/AP_AdvancedFailsafe/AP_AdvancedFailsafe.cpp]: The name below now doesn't mean much.

File: libraries/AP_AdvancedFailsafe/AP_AdvancedFailsafe.h [hunk 3]
```diff
@@ -64,19 +64,28 @@ class AP_AdvancedFailsafe
...
     // for holding parameters
     static const struct AP_Param::GroupInfo var_info[];
 
-private:
+    // called to set all outputs to termination state
+    virtual void terminate_vehicle(void) = 0;
+        
+protected:
+    // setup failsafe values for if FMU firmware stops running
+    virtual void setup_IO_failsafe(void) = 0;
```
Comment (lucasdemarchi) [inline libraries/AP_AdvancedFailsafe/AP_AdvancedFailsafe.h]: missing underscore on protected method names?

Comment (rmackay9) [inline libraries/AP_AdvancedFailsafe/AP_AdvancedFailsafe.h]: do we generally put underscores on methods?  I know we do that for members but we don't generally for methods.

Comment (lucasdemarchi) [inline libraries/AP_AdvancedFailsafe/AP_AdvancedFailsafe.h]: I thought so.

Comment (tridge) [inline libraries/AP_AdvancedFailsafe/AP_AdvancedFailsafe.h]: I've been treating this as a per-class decision. Personally I find the underscores make the code significantly harder to read, and I have not been using them for new classes that I write for either methods or variables.

File: libraries/AP_AdvancedFailsafe/AP_AdvancedFailsafe.cpp (hunk 2 above)
Comment (tridge) [inline libraries/AP_AdvancedFailsafe/AP_AdvancedFailsafe.cpp]: I've expanded the comment to make it clearer

**Comment by tridge** [comment]
applied

**Pull request closed without merging**